services/blockstorage/sync/block_sync.go

```diff
@@ -37,22 +38,51 @@ type BlockSyncStorage interface {
 	UpdateConsensusAlgosAboutLatestCommittedBlock(ctx context.Context)
 }
 
+type blockSyncConduit struct {
+	idleReset chan struct{}
+	responses chan *gossipmessages.BlockAvailabilityResponseMessage
+	blocks    chan *gossipmessages.BlockSyncResponseMessage
+}
+
 type BlockSync struct {
 	logger       log.BasicLogger
 	sf           *stateFactory
 	gossip       gossiptopics.BlockSync
 	storage      BlockSyncStorage
 	config       blockSyncConfig
 	currentState syncState
+	c            *blockSyncConduit
+
+	metrics *stateMachineMetrics
 }
 
-func NewBlockSync(ctx context.Context, config blockSyncConfig, gossip gossiptopics.BlockSync, storage BlockSyncStorage, logger log.BasicLogger) *BlockSync {
+type stateMachineMetrics struct {
+	statesTransitioned *metric.Gauge
```

Comment:
I think a good question would be "what are you trying to measure?"

Reply:
"is it even working"
also
"comparing two nodes with similar uptime should have similar values"

given that 99.999% are idle state changes (on each block committed by consensus, we shift from idle to idle)